x-pack/plugins/security_solution/public/overview/containers/overview_risky_host_links/use_hosts_risk_score.ts

```diff
@@ -9,53 +9,53 @@ import { i18n } from '@kbn/i18n';
 import { useCallback, useEffect, useState } from 'react';
 import { useDispatch } from 'react-redux';
 
-import { useRiskyHostsComplete } from './use_risky_hosts';
 import { useAppToasts } from '../../../common/hooks/use_app_toasts';
 import { useKibana } from '../../../common/lib/kibana';
 import { inputsActions } from '../../../common/store/actions';
-import { LinkPanelListItem } from '../../components/link_panel';
-import { RISKY_HOSTS_INDEX } from '../../../../common/constants';
+
+import { HOST_RISK_SCORES_INDEX } from '../../../../common/constants';
 import { isIndexNotFoundError } from '../../../common/utils/exceptions';
+import { HostsRiskScore } from '../../../../common';
+import { useHostsRiskScoreComplete } from './use_hosts_risk_score_complete';
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
 
-export const QUERY_ID = 'risky_hosts';
+export const QUERY_ID = 'host_risk_score';
 const noop = () => {};
 
-export interface RiskyHost {
-  host: {
-    name: string;
-  };
-  risk_score: number;
-  risk: string;
-}
-
 const isRecord = (item: unknown): item is Record<string, unknown> =>
   typeof item === 'object' && !!item;
 
-const isRiskyHostHit = (item: unknown): item is RiskyHost =>
+const isHostsRiskScoreHit = (item: unknown): item is HostsRiskScore =>
   isRecord(item) &&
   isRecord(item.host) &&
   typeof item.host.name === 'string' &&
   typeof item.risk_score === 'number' &&
   typeof item.risk === 'string';
 
-const getListItemsFromHits = (items: RiskyHost[]): LinkPanelListItem[] => {
-  return items.map(({ host, risk_score: count, risk: copy }) => ({
-    title: host.name,
-    count,
-    copy,
-    path: '',
-  }));
-};
+export interface HostRisk {
+  loading: boolean;
+  isModuleEnabled?: boolean;
+  result?: HostsRiskScore[];
+}
 
-export const useRiskyHostLinks = ({ to, from }: { to: string; from: string }) => {
-  const [isModuleEnabled, setIsModuleEnabled] = useState<boolean | undefined>(undefined);
+export const useHostsRiskScore = ({
```

Comment:
does `Hosts` need to be plural? it's the risk score of a single host right?

Reply:
After the refactor, we use the same query and hook for fetching the list of hosts risks on the overview page and the single host risk on the alert flyout.